util/GameRules.cpp

```diff
@@ -146,9 +146,10 @@ namespace {
             rule.validator = std::make_unique<Validator<T>>();
         auto option_name = "setup.rules." + rule.name;
         if (!GetOptionsDB().OptionExists(option_name))
-            GetOptionsDB().Add<T>(std::move(option_name), rule.description,
+            GetOptionsDB().Add<T>(option_name, rule.description,
                                   boost::any_cast<T>(rule.default_value),
                                   rule.validator->Clone());
+        rule.value = GetOptionsDB().Get<T>(option_name);
```

Comment:
Does this always need to be set, or just if there was already an existing option for that option name? If the latter, would be clearer to put it in an `else` case...

Reply:
@geoffthemedio It should be set because even if options was already registered its value could be different.

Reply:
Added comment about modification.

Reply:
My point was that if there was no option yet registered, then is there any need to set the rule value from the newly-registered option?